crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/entity/DeadLetterMessageEntity.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.laa.crime.crowncourt.entity;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
+import java.time.LocalDateTime;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "DEAD_LETTER_MESSAGE", schema = "crown_court_proceeding")
+public class DeadLetterMessageEntity {
+    @Id
+    @SequenceGenerator(name = "dead_letter_seq", sequenceName = "DEAD_LETTER", allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "dead_letter_seq")
+    @Column(name = "ID")
+    private Integer id;
+    @Column(name = "MESSAGE")
+    private byte[] message;
```

Comment:
Just to check here, which JSON type are you referring to exactly, and in the code, the DB, or both?

Reply:
Sorry, I'm suggesting to the the `jsonb` column type in Postgres. Then on the Java side, you can create a Jackson `AttributeConverter` to map between the JSON and the concrete type, in this case `ProsecutionConcluded`.

```java
Column(name = "MESSAGE", columnDefinition = "jsonb")
@Convert(converter = JsonToProsecutionConcludedConverter.class)
private ProsecutionConcluded message;
```

Reply:
Thanks for clarifying, makes sense 👍

Reply:
One thought: in the situation where the MAAT id is not valid (is anything other than an int), we're not going to be able to construct an instance of _ProsecutionConcluded_ because the type isn't going to map, which means that we can't 100% use a converter class to map to the concrete type.

Reply:
Yes, something like that should work, or we just decide not to store the messages with malformed MAAT_IDs since this appears to be the expected behaviour based on Ethan's recent email